packages/oui-datagrid/src/filter/filter.spec.js

```diff
@@ -0,0 +1,399 @@
+                });
+            });
+
+            describe("and 'contains' operator", () => {
+                const expectedResults = 2;
+
+                it("should filter", () => {
+                    filter = new Filter([], columns);
+
+                    const filtered = filter.applyCriteria(fakeData, {
+                        property: "firstName",
+                        operator: "contains",
+                        value: "aaron"
+                    });
+                    expect(filtered.length).toBe(expectedResults);
+                });
+
+                it("should filter (negated)", () => {
+                    filter = new Filter([], columns);
+
+                    const filtered = filter.applyCriteria(fakeData, {
+                        property: "firstName",
+                        operator: "containsNot",
+                        value: "aaron"
+                    });
+                    expect(filtered.length).toBe(fakeData.length - expectedResults);
+                });
+            });
+
+            describe("and 'starts with' operator", () => {
+                const expectedResults = 5;
+
+                it("should filter", () => {
+                    filter = new Filter([], columns);
+
+                    const filtered = filter.applyCriteria(fakeData, {
+                        property: "firstName",
+                        operator: "startsWith",
+                        value: "ron"
+                    });
+                    expect(filtered.length).toBe(expectedResults);
+                });
+
+                it("should filter (negated)", () => {
+                    filter = new Filter([], columns);
+
+                    const filtered = filter.applyCriteria(fakeData, {
+                        property: "firstName",
+                        operator: "startsWithNot",
+                        value: "ron"
+                    });
+                    expect(filtered.length).toBe(fakeData.length - expectedResults);
+                });
+            });
+
+            describe("and 'ends with' operator", () => {
+                const expectedResults = 23;
+
+                it("should filter", () => {
+                    filter = new Filter([], columns);
+
+                    const filtered = filter.applyCriteria(fakeData, {
+                        property: "lastName",
+                        operator: "endsWith",
+                        value: "man"
+                    });
+                    expect(filtered.length).toBe(expectedResults);
+                });
+
+                it("should filter (negated)", () => {
+                    filter = new Filter([], columns);
+
+                    const filtered = filter.applyCriteria(fakeData, {
+                        property: "lastName",
+                        operator: "endsWithNot",
+                        value: "man"
+                    });
+                    expect(filtered.length).toBe(fakeData.length - expectedResults);
```

Comment:
Nice coverage!!
(optional)Tests with fake data in an external file complicate understanding.
I do not know the meaning of expectedResults = 23 just by reading the test.
A table with 2 occurrences and one that ends with "man" could be enough.